### PR TITLE
(#37) Pager on Disease DLP

### LINKED ISSUE #37: Story: Pager on disease DLP
## As a user, I want pager so that I can look through all of the results on dynamic listing pages.

Basically, add the pager to dynamic listing pages once code is complete for manual pages.

> **ESTIMATE** 3

### Acceptance Criteria

```gherkin
Scenario: Pager on dynamic listing page results with disease parameter only
  Given the user navigates to "/C4872"
  And "trialListingPageType" is set to "Disease"
    Then the page title is "Breast Cancer Clinical Trials"
   Then the system displays "Trials 1-25 of" "912"
    And the top pager is aligned to right on the same line as the number of trials on desktop
   And the bottom pager is aligned to right
    And pager displays the following navigation options
      | pages  |
      | 1      |
      | 2      |
      | 3      |
      | ...    |
      | 37     |
      | Next > |
    And the page "1" is highlighted
```

```gherkin
Scenario: Mobile and Tablet Pager Display
  Given user navigates to a listing page results URL
  Then the system returns the results page for cancer
    Then a horizontal rule is displayed below the "Trials 1-25 of XXX" 
    Then a top pager is displayed aligned to the left
    Then a horizontal rule is displayed below the pager
    Then the result items are displayed 
    Then a horizontal rule is displayed 
    Then the bottom pager is displayed aligned to the left
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Add Pager to `Disease`
- [ ] Update unit test for `Disease` to capture Pager
- [ ] Add integration tests

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alerosad**
Closes #37

* Added Pager functionality to Disease DLP
* Updated unit tests for Disease DLP
* Updated integration tests for Disease DLP

### CODE REVIEW

File: cypress/integration/AppAnalytics/DiseaseAnalytics.feature [hunk 1]
```diff
@@ -4,21 +4,21 @@ Feature: Disease listing page analytics
...
-            | page.additionalDetails.trialListingPageType | disease					                                           |
+						| page.name                                   | www.cancer.gov/C4872                                       |
+					  | page.title                                  | Breast Cancer Clinical Trials     							           |
+						| page.channel                                | Clinical Trials                                            |
+					  | page.type                                   | nciAppModulePage                                           |
+					  | page.language                               | english                                                    |
+						| page.metaTitle                              | Breast Cancer Clinical Trials - National Cancer Institute  |
 						| page.additionalDetails.diseaseName			    | breast cancer		                                           |
+            | page.additionalDetails.trialListingPageType | disease					                                           |
+						| page.additionalDetails.numberResults        | (int)938                                                   |
```
Comment (kate-mashkina) [inline cypress/integration/AppAnalytics/DiseaseAnalytics.feature:24]: You need to add at least one integration test for a pager inside DiseasePrettyUrl.feature ( no need to have full functional test, just take the first scenario from AdultMetastaticBrainTumor.feature and implement it for breast-cancer path)

**Comment by kate-mashkina** [comment]
@mworrest @VictoriaSunNIH @alinai https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-42/breast-cancer?cfg=0

File: cypress/integration/AppAnalytics/DiseaseAnalytics.feature (hunk 1 above)
Comment (alerosad) [reply cypress/integration/AppAnalytics/DiseaseAnalytics.feature:24]: Will do

**Comment by alerosad** [comment]
> @mworrest @VictoriaSunNIH @alinai https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-42/breast-cancer?cfg=0

Thanks Kate, forgot to include the link, my bad.

File: cypress/integration/DiseasePrettyURLName.feature [hunk 2]
```diff
@@ -27,3 +27,52 @@ Feature: As a user, I would like to view the trial results for a disease listing
       | name        | content                                 																																				                          |
       | description | NCI supports clinical trials studying new and more effective ways to detect and treat cancer. Find clinical trials for breast cancer.     |
     And there is a canonical link with the href "https://www.cancer.gov/breast-cancer"
+
+
+	Scenario: Page Load Analytics fires when a user views a disease listing page
```
[the remark this answers is not in the document]
Comment (alerosad) [reply cypress/integration/DiseasePrettyURLName.feature:32]: Will do

**Review by seyilonge-nci — CHANGES_REQUESTED** [review]

**Review by seyilonge-nci — APPROVED** [review]

File: cypress/integration/common/index.js [hunk 3]
```diff
@@ -170,6 +170,18 @@ And(
...
 	}
 );
+
+/*
+    -----------------------
+        Manual Page results
+    -----------------------
+*/
+And('the page goes into mobile view', () => {
+	cy.viewport(606,498);
```
Comment (kate-mashkina) [inline cypress/integration/common/index.js:180]: I promise this is the last thing - can you make this step more generic? You can just use this exact code we have in our regression repo instead: https://github.com/NCIOCPL/cgov-digital-platform-acceptance-tests/blob/master/cypress/integration/common/config.js

**Review by kate-mashkina — CHANGES_REQUESTED** [review]

**Review by kate-mashkina — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**